packages/calcite-components/src/components/flow/flow.tsx

```diff
@@ -195,7 +248,7 @@ export class Flow implements LoadableComponent {
     };
 
     return (
-      <div class={frameDirectionClasses}>
+      <div class={frameDirectionClasses} onAnimationEnd={this.resetFlowDirection}>
```

Comment:
There are some scenarios that will prevent this event from firing (see [example](https://codepen.io/jcfranco/pen/abeYvpY?editors=1111)), so you might need to use the [`whenAnimationDone` util](https://github.com/Esri/calcite-design-system/blob/dev/packages/calcite-components/src/utils/dom.ts#L689) for consistent event handling.